From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Fri, 1 Jun 2018 17:07:41 +0200
Subject: libgdm: Unref the manager propagated from task

This instance has already been reffed when passed to the task, and since
we're stealing it with `g_task_propagate_pointer` it won't be unreffed.

We could also do this in the `on_reauthentication_channel_opened` callback
but since the new task will ref it anyway, we can just be clean and do it
here.
---
 libgdm/gdm-client.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libgdm/gdm-client.c b/libgdm/gdm-client.c
index f327344..fa4ba84 100644
--- a/libgdm/gdm-client.c
+++ b/libgdm/gdm-client.c
@@ -390,7 +390,7 @@ on_got_manager_for_reauthentication (GdmClient           *client,
                                                         (GAsyncReadyCallback)
                                                         on_reauthentication_channel_opened,
                                                         task);
-
+        g_object_unref (manager);
 }
 
 static GDBusConnection *
@@ -527,6 +527,8 @@ on_got_manager_for_opening_connection (GdmClient           *client,
                                        (GAsyncReadyCallback)
                                        on_session_opened,
                                        task);
+
+        g_object_unref (manager);
 }
 
 static GDBusConnection *
